### PR TITLE
docs - pxf jdbc partition range improvements

### PR DESCRIPTION
**Pull request by lisakowen**
enhance the JDBC partitioning doc content based on changes in https://github.com/greenplum-db/pxf/pull/170.  there was alot of discussion in this PR, and some of it related to what i think are possible future features, so let me know if i missed anything.

doc review site link:  http://docs-lisa-review.cfapps.io/6-0/pxf/jdbc_pxf.html#partitioning

question:  the last paragraph of the partitioning section discusses tuning of the INTERVAL property that seems to conflict with some of the discussion in the original PR.  can someone please review that and let me know if/how we might want to change the content?

### CODE REVIEW

**Comment by lisakowen** [comment]
@leskin-in , please review if interested.  thanks!

File: gpdb-doc/markdown/pxf/jdbc_pxf.html.md.erb [hunk 1]
```diff
@@ -144,18 +144,31 @@ The PXF JDBC connector supports simultaneous read access from PXF instances runn
...
 Example JDBC \<custom-option\> substrings that identify partitioning parameters:
 
 ``` pre
+&PARTITION_BY=id:int&RANGE=1:100&INTERVAL=5
 &PARTITION_BY=year:int&RANGE=2011:2013&INTERVAL=1
 &PARTITION_BY=createdate:date&RANGE=2013-01-01:2016-01-01&INTERVAL=1:month
 &PARTITION_BY=color:enum&RANGE=red:yellow:blue
 ```
 
+When you enable partitioning, the PXF JDBC connector splits a `SELECT` query into a set of small queries, each of which is called a fragment. The JDBC connector automatically adds extra query constraints (`WHERE` expressions) to each fragment to guarantee that every tuple of data is retrieved from the external database exactly once.
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/jdbc_pxf.html.md.erb]: I don't really like the terminology here `into a set of small queries`. What is a small query?

Comment (dyozie) [inline gpdb-doc/markdown/pxf/jdbc_pxf.html.md.erb]: small -> smaller

Comment (frankgh) [reply gpdb-doc/markdown/pxf/jdbc_pxf.html.md.erb]: How about this:
```suggestion
When you enable partitioning, the PXF JDBC connector splits a `SELECT` query into multiple subqueries that retrieve a subset of the data, each of which is called a fragment. The JDBC connector automatically adds extra query constraints (`WHERE` expressions) to each fragment to guarantee that every tuple of data is retrieved from the external database exactly once.
```

File: gpdb-doc/markdown/pxf/jdbc_pxf.html.md.erb [hunk 2]
```diff
@@ -144,18 +144,31 @@ The PXF JDBC connector supports simultaneous read access from PXF instances runn
...
+
+For example, when a user `SELECT`s a PXF external table created with a `LOCATION` clause that specifies `&PARTITION_BY=id:int&RANGE=1:5&INTERVAL=2`, PXF generates 5 fragments: two according to the partition settings and up to three implicitly generated fragments. The constraints associated with each fragment are as follows:
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/jdbc_pxf.html.md.erb]: `SELECT`s -> queries

File: gpdb-doc/markdown/pxf/jdbc_pxf.html.md.erb [hunk 3]
```diff
@@ -94,9 +94,9 @@ You include JDBC connector custom options in the `LOCATION` URI, prefacing each
...
+| PARTITION_BY | Read | Enables read partitioning. The partition column, \<column-name\>:\<column-type\>. You may specify only one partition column. The JDBC connector supports `date`, `int`, and `enum` \<column-type\> values. If you do not identify a `PARTITION_BY` column, a single PXF instance services the read request. |
+| RANGE | Read | Required when `PARTITION_BY` is specified. The query range; its format is dependent upon the data type of the partition column. When the partition column is an `enum` type, `RANGE` must specify a list of values, \<value\>:\<value\>[:\<value\>[...]], each of which forms its own fragment. If the partition column is an `int` or `date` type, `RANGE` must specify \<start-value\>:\<end-value\> and represents the closed interval from \<start-value\> through \<end-value\>, inclusive. If the partition column is a `date` type, use the `yyyy-MM-dd` date format. |
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/jdbc_pxf.html.md.erb]: Is RANGE a closed interval? We now include all values for the partition column, the range only helps create the partitions, but all data is guaranteed to be in the resulting dataset

**Pull request merged**